src/producer/base_producer.rs

```diff
@@ -494,26 +505,27 @@ where
     // As this library uses the rdkafka Event API, flush will not call rd_kafka_poll() but instead wait for
     // the librdkafka-handled message count to reach zero. Runs until value reaches zero or timeout.
     fn flush<T: Into<Timeout>>(&self, timeout: T) -> KafkaResult<()> {
-        let mut timeout = timeout.into();
-        loop {
-            let op_timeout = std::cmp::min(timeout, self.min_poll_interval);
-            if self.in_flight_count() > 0 {
-                unsafe { rdsys::rd_kafka_flush(self.native_ptr(), 0) };
-                self.poll(op_timeout);
+        let mut remaining = if let Timeout::After(dur) = timeout.into() {
+            dur
```

Comment:
if API requires i32 then we need to check it here too. What if `dur` will be e.g. `i32::MAX + 1` ?

Reply:
Fixed in 3f2b5b91.